scan-command/src/main/java/io/ballerina/scan/internal/CoreRule.java

```diff
@@ -30,9 +30,18 @@
  * @since 0.1.0
  * */
 enum CoreRule {
+
     AVOID_CHECKPANIC(RuleFactory.createRule(1, "Avoid checkpanic", RuleKind.CODE_SMELL)),
     UNUSED_FUNCTION_PARAMETER(RuleFactory.createRule(2,
-            "Unused function parameter", RuleKind.CODE_SMELL));
+            "Unused function parameter", RuleKind.CODE_SMELL)),
+    OPERATION_ALWAYS_EVALUATE_TO_TRUE(RuleFactory.createRule(6,
+            "This operation always evaluate to true", RuleKind.CODE_SMELL)),
+    OPERATION_ALWAYS_EVALUATE_TO_FALSE(RuleFactory.createRule(7,
+            "This operation always evaluate to false", RuleKind.CODE_SMELL)),
+    OPERATION_ALWAYS_EVALUATE_TO_SELF_VALUE(RuleFactory.createRule(8,
+            "This operation always evaluate to the same value", RuleKind.CODE_SMELL)),
+    SELF_ASSIGNMENT(RuleFactory.createRule(9,
+            "Self assignment", RuleKind.CODE_SMELL));
```

Comment:
Shall we change something like `Self assignment` to match the other description of other rules. 
Suggestions - 
1. This variable is self assigned
2. This variable is assigned to itself